### PR TITLE
PacketFence logs in the dashboard

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Added logs file parser in the dashboard

# Impacts
Dashboard

# Delete branch after merge
YES

# Checklist
- [n/a ] Document the feature
- [n/a ] Add unit tests
- [n/a ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Added packetfence logs graph in the dashboard

### CODE REVIEW

**Comment by extrafu** [comment]
This should be reworked to only work in the v9 GUI.

File: html/pfappserver/root/static.alt/src/views/Status/_components/Dashboard.vue [hunk 1]
```diff
@@ -601,7 +601,96 @@ export default {
               })
             }
           ] // groups
-        } // Queue section
+        }, // Queue section
+        {
+          name: this.$i18n.t('Logs'),
+          groups: [
+            {
+              name: this.$i18n.t('packetfence.log'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Status/_components/Dashboard.vue]: Don't translate filenames (change it everywhere).

File: html/pfappserver/root/static.alt/src/views/Status/_components/Dashboard.vue [hunk 2]
```diff
@@ -601,7 +601,96 @@ export default {
...
+              items: [
+                {
+                  title: this.$i18n.t('Numbers of events'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Status/_components/Dashboard.vue]: *Numbers of events* → *Number of events* (change it everywhere).

File: html/pfappserver/root/graph/dashboard.tt [hunk 3]
```diff
@@ -568,9 +569,79 @@ END;
...
           </div>
-
+        <div class="tab-pane[% IF tab == "logs" %] active[% END %]" id="logs">
+          <div class="card">
+            <div class="card-title">
+              <h2>[% l('packetfence.log') %]</h2>
+            </div>
+              <div class="card-block">
+                <div class="row-flex">
+                    [% graph("packetfence.logs.packetfence_log", "Numbers of events") | none %]
```
Comment (cgx) [inline html/pfappserver/root/graph/dashboard.tt]: *Numbers of events* → *Number of events* (change it everywhere)

File: conf/stats.conf.defaults [hunk 4]
```diff
@@ -202,3 +202,51 @@ api_compile=$.items[*].network, $.items[*].percentused
...
+match=ERROR,httpd.portal.*Unable to match MAC address to IP.*ip2mac.*
+statsd_ns=source.packetfence.packetfence_log.error,source.packetfence.packetfence_log.portal.ip2mac
+
+[metric 'DHCP messages in pfdhcp.log']
+type=tail_file
+file=/usr/local/pf/logs/pfdhcp.log
+match=DHCPDISCOVER,DHCPOFFER,DHCPREQUEST,DHCPACK,DHCPNAK,DHCPDECLINE,DHCPRELEASE,DHCPINFORM
+statsd_ns=source.packetfence.pfdhcp_log.dhcpdiscover,source.packetfence.pfdhcp_log.dhcpoffer,source.packetfence.pfdhcp_log.dhcprequest,source.packetfence.pfdhcp_log.dhcpack,source.packetfence.pfdhcp_log.dhcpnak,source.packetfence.pfdhcp_log.dhcpdecline,source.packetfence.pfdhcp_log.dhcprelease,source.packetfence.pfdhcp_log.dhcpinform
+
+[metric 'RADIUS load_balancer.log']
```
Comment (cgx) [inline conf/stats.conf.defaults]: To be more precise, can we call it *Errors in radius-load_balancer.log*?

File: conf/stats.conf.defaults [hunk 5]
```diff
@@ -202,3 +202,51 @@ api_compile=$.items[*].network, $.items[*].percentused
...
+type=tail_file
+file=/usr/local/pf/logs/radius-load_balancer.log
+match=unknown client,invalid Message-Authenticator
+statsd_ns=source.packetfence.radius_load_balancer_log.unknown_client,source.packetfence.radius_load_balancer_log.invalid_message_authenticator
+
+[metric 'RADIUS radius.log']
```
Comment (cgx) [inline conf/stats.conf.defaults]: To be more precise, can we call it *Errors in radius.log*?

File: conf/stats.conf.defaults [hunk 6]
```diff
@@ -202,3 +202,51 @@ api_compile=$.items[*].network, $.items[*].percentused
...
+type=tail_file
+file=/usr/local/pf/logs/radius.log
+match=unknown client,invalid Message-Authenticator,0xc0000072,c000006d,0xc0000001,sql_reject,Ignoring duplicate packet
+statsd_ns=source.packetfence.radius_log.unknown_client,source.packetfence.radius_log.invalid_message_authenticator,source.packetfence.radius_log.account_disabled,source.packetfence.radius_log.bad_username_or_password,source.packetfence.radius_log.reading_winbind_reply_failed,source.packetfence.radius_log.sql_reject,source.packetfence.radius_log.duplicate_packet
+
+[metric 'RADIUS radius-acct.log']
```
Comment (cgx) [inline conf/stats.conf.defaults]: To be more precise, can we call it *Errors in radius-acct.log*?

File: conf/stats.conf.defaults [hunk 7]
```diff
@@ -202,3 +202,51 @@ api_compile=$.items[*].network, $.items[*].percentused
...
+match=ERROR
+statsd_ns=source.packetfence.radius_acct_log.error
+
+[metric 'Mariadb mariadb_error.log']
+type=tail_file
+file=/usr/local/pf/logs/mariadb_error.log
+match=DESYNCED
+statsd_ns=source.packetfence.mariadb_error_log.desynced
+
+[metric 'Events in pfmon.log']
```
Comment (cgx) [inline conf/stats.conf.defaults:242]: To be more precise, can we call it *Errors in pfmon.log*?

File: conf/stats.conf.defaults [hunk 8]
```diff
@@ -202,3 +202,51 @@ api_compile=$.items[*].network, $.items[*].percentused
...
+type=tail_file
+file=/usr/local/pf/logs/pfmon.log
+match=ERROR,WARNING
+statsd_ns=source.packetfence.pfmon_log.error,source.packetfence.pfmon_log.warning
+
+[metric 'Events in fingerbank.log']
```
Comment (cgx) [inline conf/stats.conf.defaults:248]: To be more precise, can we call it *Errors in fingerbank.log*?

File: conf/stats.conf.defaults [hunk 9]
```diff
@@ -202,3 +202,51 @@ api_compile=$.items[*].network, $.items[*].percentused
...
+match=unknown client,invalid Message-Authenticator,0xc0000072,c000006d,0xc0000001,sql_reject,Ignoring duplicate packet
+statsd_ns=source.packetfence.radius_log.unknown_client,source.packetfence.radius_log.invalid_message_authenticator,source.packetfence.radius_log.account_disabled,source.packetfence.radius_log.bad_username_or_password,source.packetfence.radius_log.reading_winbind_reply_failed,source.packetfence.radius_log.sql_reject,source.packetfence.radius_log.duplicate_packet
+
+[metric 'RADIUS radius-acct.log']
+type=tail_file
+file=/usr/local/pf/logs/radius-acct.log
+match=ERROR
+statsd_ns=source.packetfence.radius_acct_log.error
+
+[metric 'Mariadb mariadb_error.log']
```
Comment (cgx) [inline conf/stats.conf.defaults]: To be more precise, can we call it *Synching errors in mariadb_error.log*?

**Review by cgx — CHANGES_REQUESTED** [review]

File: conf/stats.conf.defaults (hunk 4 above)
Comment (fdurand) [reply conf/stats.conf.defaults]: It's can be something else than error, can we use events instead ?

Comment (cgx) [reply conf/stats.conf.defaults]: Of course!

**Comment by fdurand** [comment]
Changes done

**Review by cgx — APPROVED** [review]

**Comment by fdurand** [comment]
Ready to merge ?

**Pull request merged**